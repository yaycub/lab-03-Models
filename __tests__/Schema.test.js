const Schema = require('../lib/Schema');

describe('Schema', () => {
  it('validates a good schema', () => {
    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number
      },
      weight: {
        type: String
      }
    });

    const dog = {
      name: 'spot',
      age: 5,
      weight: '20 lbs'
    };

    expect(schema.validate(dog)).toEqual({
      name: 'spot',
      age: 5,
      weight: '20 lbs'
    });
  });

  it('throws on a bad schema', () => {
    const schema = new Schema({
      name: {
        type: String,
        required: true
      },
      age: {
        type: Number
      },
      weight: {
        type: String
      }
    });

    const dog = {
      age: 'hi',
      weight: '20 lbs'
    };

    expect(() => schema.validate(dog)).toThrowErrorMatchingSnapshot();
  });
});
