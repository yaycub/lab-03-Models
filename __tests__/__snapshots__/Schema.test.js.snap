// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema throws on a bad schema 1`] = `"invalid schema >> Error: name is required,Error: Cannot cast >>hi<< to Number"`;
